hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java

```diff
@@ -119,32 +126,23 @@ private boolean skipUpdateStatistics() {
     }
 
     private void handleExecuteResponse(
-        ClientConnection connection,
         SqlClientResult res,
         ClientMessage message,
-        Throwable error
+        ClientInvocationFuture future
     ) {
-        if (error != null) {
-            res.onExecuteError(rethrow(error, connection));
-
-            return;
-        }
-
         SqlExecuteCodec.ResponseParameters response = SqlExecuteCodec.decodeResponse(message);
-
         HazelcastSqlException responseError = handleResponseError(response.error);
 
         if (responseError != null) {
             res.onExecuteError(responseError);
-
-            return;
+            future.completeExceptionally(responseError);
```

Comment:
You're completing an already completed future here, this test doesn't throw:
```java
        createHazelcastInstance();
        HazelcastInstance client = createHazelcastClient();
        client.getSql().execute("selectx");
```

Reply:
yeah this is to be fixed, I know

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/19612/commits/3a2ae1c5cb2a6ec857cc9a6691280a8a76b6621b 
this pr is ready to review now, sorry for inconvenience.